dataset/data_array.cpp

```diff
@@ -319,6 +319,7 @@ DataArray &DataArray::operator/=(const VariableConstView &other) {
 
 DataArray operator+(const DataArrayConstView &a, const DataArrayConstView &b) {
   if (a.hasData() && b.hasData()) {
+    expect::coordsAreSuperset(a, b);
```

Comment:
This should not be here. `coordsAreSuperset` is an assymetric check for in-place operations and in particular does not support broadcasting (in-place output cannot be larger). The check for compatible coords happens in `union_` below (and broadcasting is permitted).